flepimop/gempyor_pkg/src/gempyor/utils.py

```diff
@@ -100,6 +100,22 @@ def read_df(
 
 
 def command_safe_run(command, command_name="mycommand", fail_on_fail=True):
+    """
+    Verifies that a command is valid by attempting to run it. Prints stream of code if command fails.
+
+    Args:
+        command: The CLI command to be given (str).
+        command_name: The reference name for you command (str). Default value is "mycommand".
+        fail_on_fail: Boolean; default is True. If True, an exception will be thrown if the command fails.
+
+    Returns:
+        returncode: The returncode message from running yourcommand.
+        stdout: Standard output
+        stderr: Standard error stream
+
+    Raises:
+        Exception: If fail_on_fail=True and the command fails, an exception will be thrown.
```

Comment:
Especially for these utility functions where it is easy to use them independently examples are very helpful. Here's an example of how to add in examples: https://github.com/HopkinsIDD/flepiMoP/blob/main/flepimop/gempyor_pkg/src/gempyor/file_paths.py#L156.